water/tools/water-opt/CMakeLists.txt

```diff
@@ -11,6 +11,11 @@ set(LIBS
 
         MLIRWaterTestTransforms
         MLIRWaterTestDialect
+
+        MLIRRegisterAllDialects
+        MLIRRegisterAllExtensions
+        MLIRRegisterAllPasses
```

Comment:
Please don't. Just pick what you need. Otherwise this _significantly_ increases build times and distribution sizes, and is near impossible to undo.